backend/app/services/link_preview.py

```
@@ -0,0 +1,425 @@
+
+def _get_cache_key(url: str) -> str:
+    """Generate a cache key for link preview"""
+    url_hash = hashlib.md5(url.encode()).hexdigest()
+    return f"link_preview:{url_hash}"
+
+
+def _get_redis_client():
+    """Get Redis client instance"""
+    try:
+        return redis.from_url(settings.REDIS_URL)
+    except Exception as e:
```

Ruff (0.14.11)

Do not catch blind exception: `Exception`

(BLE001)